File: puzzles/100_Doors/rust/100_doors.rs
fn main() {
    let mut doors = vec![false; 100];

    for i in 0..100 {
        let mut j = i;
        while j < 100 {
            doors[j] = !doors[j];
            j += i + 1;
        }
    }

    for i in 0..100 {
        if doors[i] {
            println!("Door {} is open", i + 1);
        }
    }
}

File: puzzles/100_Doors/elixir/100_doors.exs
defmodule Doors do
	def run do
		(1..100)
		|> Enum.map(&({&1, :closed}))
		|> open_doors(Enum.to_list(1..100))
		|> output
	end

	def output(list) do
		list
		|> Enum.map(fn({a, b}) -> IO.inspect("Door #{a} is #{b}") end)
	end

	def open_doors(doors, []), do: doors
	def open_doors(doors, [h | t]) do
		open_doors(doors |> Enum.map(fn({door, state}) -> update_if_divisible(door, state, h) end), t)
	end

	def update_if_divisible(door, state, h) when rem(door, h) == 0 do
		case state do
			:closed ->
				{door, :open}
			:open ->
				{door, :closed}
		end
	end

	def update_if_divisible(door, state, _h), do: {door, state}
end

Doors.run()